### PR TITLE
Feature/intra process

### PR DESCRIPTION
**Pull request by MiguelBarro**
Updating **xml profiles** to include intraprocess set up:

* **INTRAPROCESS_OFF**. Default value, the feature is disabled.
* **INTRAPROCESS_USER_DATA_ONLY**. Discovery metratraffic keeps using ordinary transport.
* **INTRAPROCESS_FULL**. Discovery metatraffic avoids using ordinary transport.

Updating `discovery settings` to include new participant filtering capabilities.
* **FILTER_DIFFERENT_HOST** all metadata from another host, implying a different local network address, would be discarded.
* **FILTER_DIFFERENT_PROCESS** all metadata from another process on the same host would be discarded.
* **FILTER_SAME_PROCESS** all metadata from our own process would be discarded.
* **FILTER_DIFFERENT_PROCESS | FILTER_SAME_PROCESS** all metada from our own host would be discarded.

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/106/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/107/)

File: docs/advanced.rst [hunk 1]
```diff
@@ -58,6 +58,19 @@ This consist of defining a maximum number of data sinks and a maximum size for e
 Note that your History must be big enough to accommodate the maximum number of samples for each key.
 eProsima Fast RTPS will notify you if your History is too small.
 
+.. _intraprocess-delivery
```
Comment (MiguelCompany) [inline docs/advanced.rst]:
```suggestion
.. _intraprocess-delivery:
```

File: docs/advanced.rst [hunk 2]
```diff
@@ -58,6 +58,19 @@ This consist of defining a maximum number of data sinks and a maximum size for e
...
+
+Intraprocess delivery
+*********************
+
+*eProsima Fast RTPS* allows to speed up intraprocess communications by avoiding any copy operation involved with
+the transport layer. This feature is disable by default and must be enable using :ref:`xml-profiles`. Currently the
```
Comment (MiguelCompany) [inline docs/advanced.rst]:
```suggestion
the transport layer. This feature is disabled by default and must be enable using :ref:`xml-profiles`. Currently the
```

File: docs/advanced.rst [hunk 3]
```diff
@@ -691,6 +704,13 @@ DiscoverySettings
  |    :end-before: <!--><-->                        |
  +--------------------------------------------------+
 
++ a **ignoreParticipantFlags** member specifies participant filtering criteria to optimize discovery stage speed and memory usage. This feature is only available for the `SIMPLE` discovery protocol. There are several options:
```
Comment (MiguelCompany) [inline docs/advanced.rst]:
```suggestion
+ a **ignoreParticipantFlags** member specifies participant filtering criteria to optimize discovery stage speed
and memory usage. This feature is only available for the `SIMPLE` discovery protocol. There are several options:
```

File: docs/advanced.rst [hunk 4]
```diff
@@ -691,6 +704,13 @@ DiscoverySettings
...
+
+ * **FILTER_DIFFERENT_HOST** all metadata from another host, implying a different local network address, would be discarded.
```
Comment (MiguelCompany) [inline docs/advanced.rst]:
```suggestion
 * **FILTER_DIFFERENT_HOST** all metadata from another host would be discarded.
```

File: docs/xmlprofiles.rst [hunk 5]
```diff
@@ -817,6 +838,15 @@ This section of the :class:`Participant's rtps` configuration allows defining bu
 |                            | is set to :class:`STATIC`             |                         |                      |
 +----------------------------+---------------------------------------+-------------------------+----------------------+
 
+.. _Participantfiltering:
+
+**ignoreParticipantFlags**
+
+**FILTER_DIFFERENT_HOST** all metadata from another host, implying a different local network address, would be discarded.
```
Comment (MiguelCompany) [inline docs/xmlprofiles.rst]:
```suggestion
**FILTER_DIFFERENT_HOST** all metadata from another host would be discarded.
```

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/108/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/109/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/110/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/111/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/112/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/113/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/114/)

**Pull request merged**